java/client/src/main/java/glide/api/models/commands/FT/FTSearchOptions.java

```diff
@@ -0,0 +1,130 @@
+/** Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands.FT;
+
+import static glide.api.models.GlideString.gs;
+
+import glide.api.commands.servermodules.FT;
+import glide.api.models.GlideString;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.Builder;
+import org.apache.commons.lang3.tuple.Pair;
+
+/** Mandatory parameters for {@link FT#search}. */
+@Builder
+public class FTSearchOptions {
+
+    @Builder.Default private final Map<GlideString, GlideString> identifiers = new HashMap<>();
+
+    /** Query timeout in milliseconds. */
+    private final Integer timeout;
+
+    private final Pair<Integer, Integer> limit;
+
+    @Builder.Default private final boolean count = false;
+
+    /**
+     * Query parameters, which could be referenced in the query by <code>$</code> sign, followed by
+     * the parameter name.
+     */
+    @Builder.Default private final Map<String, GlideString> params = new HashMap<>();
```

Comment:
The key for params could be GlideString.

Reply:
Keys are defined by the module, they are pre-defined strings.
We use glide string for user data only.

Reply:
These parameter keys will occur in the query string and query parameter is user data.
If query is provided in GlideString form, won't the user provide the parameter keys as GlideString inputs as well?

Reply:
updated